ReactiveCocoa/Swift/Signal.swift

```diff
@@ -140,16 +140,25 @@ public final class Signal<T, E: ErrorType> {
 			return nil
 		}
 	}
+}
 
-	/// Observes the Signal by invoking the given callbacks when events are
-	/// received. If the Signal has already terminated, the `interrupted`
-	/// callback will be invoked immediately.
-	///
-	/// Returns a Disposable which can be used to stop the invocation of the
-	/// callbacks. Disposing of the Disposable will have no effect on the Signal
-	/// itself.
-	public func observe(next next: (T -> ())? = nil, error: (E -> ())? = nil, completed: (() -> ())? = nil, interrupted: (() -> ())? = nil) -> Disposable? {
-		return observe(Event.sink(next: next, error: error, completed: completed, interrupted: interrupted))
+public protocol SignalType {
+	/// The type of values being sent on the signal.
+	typealias T
+	/// The type of error that can occur on the signal. If errors aren't possible
+	/// than `NoError` can be used.
+	typealias E: ErrorType
+
+	/// Proxy for the actual signal.
```

Comment:
Can we maybe say something like, “extracts a signal from the receiver?”

I could see this protocol being implemented by stuff like `PropertyType`, for example, in which case “actual signal” might not be quite accurate.